python/cugraph/centrality/katz_centrality_wrapper.pyx

```diff
@@ -65,9 +65,17 @@ def katz_centrality(input_graph, alpha=0.1, max_iter=100, tol=1.0e-5, nstart=Non
 
     cdef bool has_guess = <bool> 0
     if nstart is not None:
-        cudf.bindings.copying.apply_scatter([nstart['values']._column],
-                                            nstart['vertex']._column._data.mem,
-                                            [df['katz_centrality']._column])
+        if input_graph.renumbered is True:
+            renumber_series = cudf.Series(input_graph.edgelist.renumber_map.index,
+                                          index=input_graph.edgelist.renumber_map)
+            nstart_vertex_renumbered = cudf.Series(renumber_series.loc[nstart['vertex']], dtype=np.int32)
+            cudf.bindings.copying.apply_scatter([nstart['values']._column],
+                                                nstart_vertex_renumbered._column,
+                                                [df['katz_centrality']._column])
+        else:
+            cudf.bindings.copying.apply_scatter([nstart['values']._column],
```

Comment:
We can see this as a sparse eigenvalue problem of size V. So, if nstart has less than V entries it should throw an error. Also, I don’t think 0 would be a valid guess. To me, a valid guess makes sense as a whole set of size V.

The alternative is to document the default value and type of normalization (as it probably won’t work if not normalized) when the user provided a subset of the guess. There should also be a test in this case. But this looks like a stretch / corner case to me. It is not a priority at this point.

Accepting vertices / value pairs in an arbitrary order in nstart is important though.

Reply:
I believe we should go with the first option and work on the second in further iterations(if required)

Reply:
agreed